examples/vae/train_vae.py

```diff
@@ -0,0 +1,546 @@
+    )
+    logger.info(f"  Gradient Accumulation steps = {args.gradient_accumulation_steps}")
+    global_step = 0
+    first_epoch = 0
+
+    progress_bar = tqdm(
+        range(global_step, args.max_train_steps),
+        disable=not accelerator.is_local_main_process,
+    )
+    progress_bar.set_description("Steps")
+
+    lpips_loss_fn = lpips.LPIPS(net="alex").to(accelerator.device)
+
+    for epoch in range(first_epoch, args.num_train_epochs):
+        vae.train()
+        train_loss = 0.0
+        for step, batch in enumerate(train_dataloader):
+            with accelerator.accumulate(vae):
+                target = batch["pixel_values"].to(weight_dtype)
+
+                # https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/autoencoder_kl.py
+                posterior = vae.encode(target).latent_dist
+                z = posterior.mode()
+                pred = vae.decode(z).sample
+
+                kl_loss = posterior.kl().mean()
+                mse_loss = F.mse_loss(pred, target, reduction="mean")
+                lpips_loss = lpips_loss_fn(pred, target).mean()
+
+                loss = (
+                    mse_loss + args.lpips_scale * lpips_loss + args.kl_scale * kl_loss
+                )
+
+                # Gather the losses across all processes for logging (if we use distributed training).
+                avg_loss = accelerator.gather(loss.repeat(args.train_batch_size)).mean()
+                train_loss += avg_loss.item() / args.gradient_accumulation_steps
+
+                accelerator.backward(loss)
+                optimizer.step()
+                lr_scheduler.step()
+                optimizer.zero_grad()
+
+            # Checks if the accelerator has performed an optimization step behind the scenes
+            if accelerator.sync_gradients:
+                progress_bar.update(1)
+                global_step += 1
+                accelerator.log({"train_loss": train_loss}, step=global_step)
+                train_loss = 0.0
+
+                if global_step % args.checkpointing_steps == 0:
+                    if accelerator.is_main_process:
+                        save_path = os.path.join(
+                            args.output_dir, f"checkpoint-{global_step}"
+                        )
+                        accelerator.save_state(save_path)
+                        logger.info(f"Saved state to {save_path}")
+
+            logs = {
+                "step_loss": loss.detach().item(),
+                "lr": lr_scheduler.get_last_lr()[0],
+                "mse": mse_loss.detach().item(),
+                "lpips": lpips_loss.detach().item(),
+                "kl": kl_loss.detach().item(),
+            }
+            accelerator.log(logs)
+            progress_bar.set_postfix(**logs)
+
+        if accelerator.is_main_process:
+            if epoch % args.validation_epochs == 0:
+                log_validation(test_dataloader, vae, accelerator, weight_dtype, epoch)
```

Comment:
```suggestion
         with torch.no_grad():
                log_validation(test_dataloader, vae, accelerator, weight_dtype, epoch)
```

this fixes the VRAM issue as suggested by @epitaque